src/librustc/mir/interpret/allocation.rs

```diff
@@ -244,7 +244,7 @@ impl<'tcx, Tag: Copy, Extra: AllocationExtra<Tag>> Allocation<Tag, Extra> {
                 Ok(&self.get_bytes(cx, ptr, size_with_null)?[..size])
             }
             // This includes the case where `offset` is out-of-bounds to begin with.
-            None => err!(UnterminatedCString(ptr.erase_tag())),
+            None => throw_unsup!(UnterminatedCString(ptr.erase_tag())),
```

Comment:
@RalfJung this is one of the examples where I'd not accept a `return`, so my question is why `throw` would be ok

Reply:
I don't see anything wrong with this code, TBH. I guess your taste is different.

In fact I'd probably change it to `Ok(match ... { ... })` with the new `throw_` macro.

Reply:
@saleemjaffer could you try to move the `Ok` to the outside here as I described above?

Then I'll do one final round reading over the entire thing, but I don't expect anything to come up except maybe a nit or two.

Reply:
> In fact I'd probably change it to Ok(match ... { ... }) with the new throw_ macro.

I guess you mean with the new `err_` macro. Using a `throw_` macro wrapped with an Ok would result in something like `Ok(Err(...))`

Reply:
So, I mean because this uses `throw_` inside the `match` now:
```rust
Ok(match ... { Some(size) => ..., None => throw_unsup!(...) })
```
This would not have worked before or with the new `err_*!`.

Reply:
Done.